cartography/intel/azure/function.py

```diff
@@ -13,75 +14,325 @@
-        return function_list
+        return function_app_list
 
     except HttpResponseError as e:
-        logger.warning(f"Error while retrieving functions - {e}")
+        logger.warning(f"Error while retrieving function app - {e}")
         return []
 
-def load_function(neo4j_session: neo4j.Session, subscription_id: str, function_list: List[Dict], update_tag: int) -> None:
-    ingest_fun = """
-    UNWIND {functions} AS function
-    MERGE (f:AzureFunction{id: function.id})
+def _load_function_app(tx: neo4j.Transaction, subscription_id: str, function_app_list: List[Dict], update_tag: int) -> None:
+    ingest_function_app = """
+    UNWIND {function_app_list} AS function_app
+    MERGE (f:AzureFunctionApp{id: function_app.id})
     ON CREATE SET f.firstseen = timestamp(),
-    f.type = function.type,
-    f.location = function.location,
-    f.resourcegroup = function.resource_group
+    f.type = function_app.type,
+    f.location = function_app.location,
+    f.resourcegroup = function_app.resource_group
     SET f.lastupdated = {update_tag}, 
-    f.name = function.name, 
-    f.container_size = function.container_size,
-    f.default_host_name=function.default_host_name, 
-    f.last_modified_time_utc=function.last_modified_time_utc,
-    f.state=function.state, 
-    f.repository_site_name=function.repository_site_name,
-    f.daily_memory_time_quota=function.daily_memory_time_quota,
-    f.availability_state=function.availability_state, 
-    f.usage_state=function.usage_state
+    f.name = function_app.name, 
+    f.container_size = function_app.container_size,
+    f.default_host_name=function_app.default_host_name, 
+    f.last_modified_time_utc=function_app.last_modified_time_utc,
+    f.state=function_app.state, 
+    f.repository_site_name=function_app.repository_site_name,
+    f.daily_memory_time_quota=function_app.daily_memory_time_quota,
+    f.availability_state=function_app.availability_state, 
+    f.usage_state=function_app.usage_state
     WITH f
     MATCH (owner:AzureSubscription{id: {SUBSCRIPTION_ID}})
     MERGE (owner)-[r:RESOURCE]->(f)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {update_tag}
     """
 
-    neo4j_session.run(
-        ingest_fun,
-        functions=function_list,
+    tx.run(
+        ingest_function_app,
+        function_app_list=function_app_list,
         SUBSCRIPTION_ID=subscription_id,
         update_tag=update_tag,
     )
 
 
 
 
+def cleanup_function_app(neo4j_session: neo4j.Session, common_job_parameters: Dict) -> None:
+    run_cleanup_job('azure_import_function_app_cleanup.json', neo4j_session, common_job_parameters)
+
+
+def sync_function_app(
+    neo4j_session: neo4j.Session, credentials: Credentials, subscription_id: str, update_tag: int,
+    common_job_parameters: Dict,
+) -> None:
+    function_app_list = get_function_app_list(credentials, subscription_id)
+    perform_db_action_function_app(neo4j_session,subscription_id,function_app_list,update_tag)
```

Comment:
can we rename this function (`perform_db_action_function_app`) to match with other function names?